Dashboard/app/js/lib/components/devices/AssignmentsEditView/index.jsx

```diff
@@ -0,0 +1,201 @@
+/* eslint-disable import/no-unresolved */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import FolderSurveySelectorView from 'akvo-flow/components/selectors/FolderSurveySelector';
+import FormSelectorView from 'akvo-flow/components/selectors/FormSelector';
+import DeviceGroupSelectorView from 'akvo-flow/components/selectors/DeviceSelector';
+import AssignmentDetails from './AssignmentDetails';
+
+export default class AssignmentsEditView extends React.Component {
+  state = {
+    assignmentName: this.props.inputValues.assignmentName,
+    startDate: this.props.inputValues.startDate,
+    expireDate: this.props.inputValues.toDate,
+    nameValidationMsg: '',
+  }
+
+  // lifecycle methods
+  componentDidMount() {
+    this.validateAssignment(this.props.inputValues.assignmentName);
+  }
+
+  // event handlers
+  onChangeState = (key, value) => {
+    if (key === 'assignmentName') this.validateAssignment(value);
+    this.setState({ [key]: value });
+  }
+
+  onSubmit = () => {
+    const { assignmentName, startDate, expireDate } = this.state;
+    if (this.assignmentNotComplete()) return;
+
+    this.props.actions.onSubmit({ assignmentName, startDate, expireDate });
+  }
+
+  // helpers
+  validateAssignment = (assignmentName) => {
+    if ((assignmentName && assignmentName.length > 100) || !assignmentName || assignmentName == '') {
+      if (assignmentName && assignmentName.length > 100) {
+        this.setState({ nameValidationMsg: Ember.String.loc('_assignment_name_over_100_chars') });
+      } else if (!assignmentName || assignmentName == '') {
+        this.setState({ nameValidationMsg: Ember.String.loc('_assignment_name_not_set') });
+      }
+    } else {
+      this.setState({ nameValidationMsg: '' });
+    }
+  }
+
+  formatStateForComponents = () => {
+    const assignmentDetailsState = {
+      assignmentName: this.state.assignmentName,
+      startDate: this.state.startDate,
+      expireDate: this.state.expireDate,
+    };
+
+    return {
+      assignmentDetailsState,
+    };
+  }
+
+  assignmentNotComplete = () => {
```

Comment:
This seems to be doing the same thing as the `validateAssignment` function above except it shows a dialog popup. Although the date validation below is not included but maybe we can implement that in a similar way to the validation before and then skip using the `FLOW.dialogControl`. One less dependency on the global `FLOW` variable. What do you think?

Reply:
Will be addressed under https://github.com/akvo/akvo-flow/issues/3221